src/executor/drop_executor.cpp

```diff
@@ -178,16 +183,29 @@ bool DropExecutor::DropTrigger(const planner::DropPlan &node,
   return false;
 }
 
+/**
+ * @brief   Drop an index.
+ * @details Drop an index. Will delete the corresponding entry from catalog, and
+ * the data structure of index itself will be garbage collected. Note that
+ * phantom
+ * could happen while dropping the index.
+ * @param   node    current planner node
+ * @param   txn     current transaction
+ * @return  bool    always false
+ */
 bool DropExecutor::DropIndex(const planner::DropPlan &node,
                              concurrency::TransactionContext *txn) {
+  // Get the name of index and the index in catalog.
   std::string index_name = node.GetIndexName();
   auto index_object =
       catalog::IndexCatalog::GetInstance()->GetIndexObject(index_name, txn);
 
+  // Drop index in catalog.
```

Comment:
Do we need the lock when drop the index?

Reply:
No, we don't. It's first logically deleted and then garbage collected, so no lock is required since no other new txns will be able to see the index.

Reply:
What if there's another index scan going on?

Reply:
It'll see the index anyway, since it still has a pointer to the index. The index itself will be garbage collected only after no thread is holding any pointer to the index, as it utilizes the smart pointer in C++. 
It's a little bit complicated here, actually. If you have further questions, I can explain to you in person about how this works.

Reply:
Thanks, the explanation is pretty clear!

Reply:
Thanks, the explanation is pretty clear!